code/identity/src/main/java/com/adobe/marketing/mobile/identity/IdentityExtension.java

```diff
@@ -1857,17 +1847,7 @@ void processPrivacyChange(final Event event, final Map<String, Object> eventData
                 privacyStatus.getValue());
 
         if (privacyStatus == MobilePrivacyStatus.OPT_OUT) {
-            mid = null;
-            advertisingIdentifier = null;
-            blob = null;
-            locationHint = null;
-            customerIds = null;
-
-            if (namedCollection != null) {
-                namedCollection.remove(DataStoreKeys.AID_SYNCED_KEY);
-            }
-
-            updatePushIdentifier(null);
+            clearIdentifiers();
             savePersistently(); // clear datastore
```

Comment:
Should we move savePersistently() to clearIdentifiers() as it is part of syncing cleared values to persistence?

Reply:
@kevinlind What do you think about this change?